test/src/test/java/org/corfudb/integration/LogReplicationLogicalGroupIT.java

```diff
@@ -0,0 +1,728 @@
+        int numTables = 2;
+        openFederatedTable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcFederatedTables);
+        openGroupATable(numTables, sourceCorfuStores.get(SOURCE_INDEX), srcTablesGroupA);
+        openFederatedTable(numTables, sinkCorfuStores.get(SINK1_INDEX), sinkFederatedTables);
+        openGroupATable(numTables, sinkCorfuStores.get(SINK2_INDEX), sinkTablesGroupA);
+
+        // Write data to Federated Table, Group A Table on Sources
+        for (int i = 0; i < NUM_WRITES; i++) {
+            StringKey key = StringKey.newBuilder().setKey(String.valueOf(i)).build();
+            ValueFieldTagOne federatedTablePayload = ValueFieldTagOne.newBuilder().setPayload(String.valueOf(i)).build();
+            SampleGroupMsgA groupATablePayload = SampleGroupMsgA.newBuilder().setPayload(String.valueOf(i)).build();
+
+            try (TxnContext txn = sourceCorfuStores.get(SOURCE_INDEX).txn(NAMESPACE)) {
+                for (int j = 0; j < srcFederatedTables.size(); j++) {
+                    txn.putRecord(srcFederatedTables.get(j), key, federatedTablePayload, null);
+                }
+                for (int j = 0; j < srcTablesGroupA.size(); j++) {
+                    txn.putRecord(srcTablesGroupA.get(j), key, groupATablePayload, null);
+                }
+                txn.commit();
+            }
+        }
+
+        // Open table Group B on sink
+        openGroupBTable(numTables, sinkCorfuStores.get(SINK3_INDEX), sinkTablesGroupB);
+
+        // Write local data to table Group B on sink
+        writeDataOnSink(NUM_WRITES, NUM_WRITES/2);
+
+        // Start the log replication
+        startReplicationServers();
+
+        // Verify the status of sync
+        verifyInLogEntrySyncState(SINK1_INDEX, SessionManager.getDefaultSubscriber());
+        verifyInLogEntrySyncState(SINK2_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+        verifyInLogEntrySyncState(SINK3_INDEX, SessionManager.getDefaultLogicalGroupSubscriber());
+
+        // Verify that the data is replicated
+        verifyFederatedTableDataOnSink(sinkCorfuStores.get(SINK1_INDEX), NUM_WRITES, sinkFederatedTables);
+        verifyGroupATableDataOnSink(sinkCorfuStores.get(SINK2_INDEX), NUM_WRITES, sinkTablesGroupA);
+        // Verify that the local data is overwritten after replication(Table Group B does not have data on Source)
+        verifyGroupBTableDataOnSink(sinkCorfuStores.get(SINK3_INDEX), 0, sinkTablesGroupB);
+
+        // Verify that the local data is overwritten after replication
+        for (int i = NUM_WRITES; i < NUM_WRITES + NUM_WRITES/2 ; i++) {
+            Sample.StringKey key = Sample.StringKey.newBuilder().setKey(String.valueOf(i)).build();
+            try (TxnContext txn = sinkCorfuStores.get(SINK1_INDEX).txn(NAMESPACE)) {
+                assertThat(txn.getRecord(sinkFederatedTables.get(SINK1_INDEX), key).getPayload()).isNull();
+                txn.commit();
+            }
+        }
```

Comment:
What is verified here? Sink index doesn't have any semantics with sinkFederatedTables right?

Reply:
I was trying to verify if the local data was overwritten after the log replication. For some reasons I mixed up the SINK1_INDEX and the index of sinkFederatedTables. Just changed it to 
`for (int i = NUM_WRITES; i < NUM_WRITES + NUM_WRITES/2 ; i++) {
            Sample.StringKey key = Sample.StringKey.newBuilder().setKey(String.valueOf(i)).build();
            try (TxnContext txn = sinkCorfuStores.get(SINK1_INDEX).txn(NAMESPACE)) {
                for (int j = 0; j < sinkFederatedTables.size(); j++) {
                    assertThat(txn.getRecord(sinkFederatedTables.get(j), key).getPayload()).isNull();
                }
                txn.commit();
            }
        }`